lib/warrant/models/warrant.rb

```diff
@@ -103,6 +104,38 @@ def self.list(filters = {})
             end
         end
 
+        # Query to find all warrants for a given subject.
+        #
+        # @option params [String] :object_type The type of object. Must be one of your system's existing object types. (optional)
+        # @option params [String] :relation The relation for this object to subject association. The relation must be valid as per the object type definition. (optional)
+        # @option params [String] :subject The subject to query warrants for. This should be in the format `OBJECT_TYPE:OBJECT_ID`, i.e. `user:8`
+        #   * subject (Hash) - The specific subject for which warrants will be queried for.
+        #       * object_type (String) - The type of object. Must be one of your system's existing object types.
+        #       * object_id (String) - The id of the specific object.
+        #
+        # @return [Array<Warrant>] list of all warrants with provided params
+        #
+        # @raise [Warrant::InternalError]
+        # @raise [Warrant::InvalidRequestError]
+        # @raise [Warrant::NotFoundError]
+        # @raise [Warrant::UnauthorizedError]
+        # @raise [Warrant::WarrantError]
+        def self.query(params = {})
+            params[:subject] = Subject.new_from_hash(params[:subject])
+            res = APIOperations.get(URI.parse("#{::Warrant.config.api_base}/v1/query"), params)
+
+            case res
+            when Net::HTTPSuccess
+                warrants = JSON.parse(res.body)
+                warrants.map{ |warrant|
+                    subject = Subject.new(warrant['subject']['objectType'], warrant['subject']['objectId'])
```

Comment:
With the new `new_from_hash` method on `Subject`, can this line now be simplified?
```ruby
subject = Subject.new(warrant['subject'])
```